client/settings/fraud-protection/advanced-settings/index.tsx

```diff
@@ -72,28 +78,32 @@ const Breadcrumb = () => (
 	</h2>
 );
 
-const SaveFraudProtectionSettingsButton = ( { children } ) => {
+const SaveFraudProtectionSettingsButton: React.FC = ( { children } ) => {
 	const headerElement = document.querySelector(
 		'.woocommerce-layout__header-wrapper'
 	);
 	return headerElement && ReactDOM.createPortal( children, headerElement );
 };
 
-const FraudProtectionAdvancedSettingsPage = () => {
-	const { saveSettings, isLoading, isSaving } = useSettings();
+const FraudProtectionAdvancedSettingsPage: React.FC = () => {
+	const { saveSettings, isLoading, isSaving } = useSettings() as UseSettings;
 
-	const cardObserver = useRef( null );
+	const cardObserver = useRef< IntersectionObserver | null >( null );
```

Comment:
I tried removing ` | null` but when I did, I got the following error on line 216:

> Cannot assign to 'current' because it is a read-only property

```TypeScript
cardObserver.current = new IntersectionObserver( observerCallback );
```

So I left it as is.

Reply:
That's right, the way to do it is to use `undefined` instead of `null`. Defining it like `useRef< IntersectionObserver >();`